# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/win_wlanLibrary.h
# Commit: afd62e82fe83
# Full Hash: afd62e82fe8320776e9696e3a8881eaae56b8822
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-11 04:06:39
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 9c94ab8b34c0 -r afd62e82fe83 netwerk/wifi/win_wlanLibrary.h
--- a/netwerk/wifi/win_wlanLibrary.h	Wed May 10 19:29:32 2023 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,54 +0,0 @@
-/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim:set ts=2 sts=2 sw=2 et cin: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#pragma once
-
-// Moz headers (alphabetical)
-
-// System headers (alphabetical)
-#include <windows.h>  // HINSTANCE, HANDLE
-#include <wlanapi.h>  // Wlan* functions
-
-class WinWLANLibrary {
- public:
-  static WinWLANLibrary* Load();
-  ~WinWLANLibrary();
-
-  HANDLE GetWLANHandle() const;
-  decltype(::WlanEnumInterfaces)* GetWlanEnumInterfacesPtr() const;
-  decltype(::WlanGetNetworkBssList)* GetWlanGetNetworkBssListPtr() const;
-  decltype(::WlanFreeMemory)* GetWlanFreeMemoryPtr() const;
-  decltype(::WlanCloseHandle)* GetWlanCloseHandlePtr() const;
-  decltype(::WlanOpenHandle)* GetWlanOpenHandlePtr() const;
-  decltype(::WlanRegisterNotification)* GetWlanRegisterNotificationPtr() const;
-  decltype(::WlanScan)* GetWlanScanPtr() const;
-
- private:
-  WinWLANLibrary();
-  bool Initialize();
-
-  HMODULE mWlanLibrary;
-  HANDLE mWlanHandle;
-  decltype(::WlanEnumInterfaces)* mWlanEnumInterfacesPtr;
-  decltype(::WlanGetNetworkBssList)* mWlanGetNetworkBssListPtr;
-  decltype(::WlanFreeMemory)* mWlanFreeMemoryPtr;
-  decltype(::WlanCloseHandle)* mWlanCloseHandlePtr;
-  decltype(::WlanOpenHandle)* mWlanOpenHandlePtr;
-  decltype(::WlanRegisterNotification)* mWlanRegisterNotificationPtr;
-  decltype(::WlanScan)* mWlanScanPtr;
-};
-
-class ScopedWLANObject {
- public:
-  ScopedWLANObject(const WinWLANLibrary& library, void* object)
-      : mLibrary(library), mObject(object) {}
-
-  ~ScopedWLANObject() { (*(mLibrary.GetWlanFreeMemoryPtr()))(mObject); }
-
- private:
-  const WinWLANLibrary& mLibrary;
-  void* mObject;
-};